bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -677,6 +678,43 @@ impl Connection {
         unsafe { Ok(Some(std::slice::from_raw_parts(chain, len as usize))) }
     }
 
+    /// The memory backing the ClientHello is owned by the Connection, so we
+    /// tie the ClientHello to the lifetime of the Connection. This is validated
+    /// with a doc test that ensures the ClientHello is invalid once the
+    /// connection has gone out of scope.
```

Comment:
I'm rusty on my Rust, but this comment kind of feels like overkill. Isn't this basically explaining standard Rust borrow behavior?

Reply:
I'm gonna say no. If we were working with safe Rust, then yes. But in this case, we're working with pointers that don't have many guarantees. This comment is my argument for why it is safe and correct to treat this pointer as a reference that is tied to the lifetime of Connection.